src/gsi/setupdbz.f90

```diff
@@ -360,97 +616,604 @@ subroutine setupdbz(obsLL,odiagLL,lunin,mype,bwork,awork,nele,nobs,is,radardbz_d
+                 else if ( i_melt_graupel  == 100 ) then
+!                   melting: wet graupel at any temperature
+                    jqg_num = Cg_wet*((rho)**(Pg_wet))*(Pg_wet)*((qgexp**(Pg_wet - one)))
+                 else
+!                   melting: depending on the temperature
+                    if (T1D < (273.15_r_kind - 2.5_r_kind)) then
+                       jqg_num = Cg_dry*((rho)**(Pg_dry))*(Pg_dry)*((qgexp**(Pg_dry - one)))
+                    else if (T1D > (273.15_r_kind + 2.5_r_kind)) then
+                       jqg_num = Cg_wet*((rho)**(Pg_wet))*(Pg_wet)*((qgexp**(Pg_wet - one)))
+                    else
+                       wgt_dry = abs(T1D - (273.15_r_kind + 2.5_r_kind))/5.0_r_kind
+                       wgt_wet = abs(T1D - (273.15_r_kind - 2.5_r_kind))/5.0_r_kind
+                       jqg_num_dry = Cg_dry*((rho)**(Pg_dry))*(Pg_dry)*((qgexp**(Pg_dry - one)))
+                       jqg_num_wet = Cg_wet*((rho)**(Pg_wet))*(Pg_wet)*((qgexp**(Pg_wet - one)))
+                       jqg_num = wgt_dry*jqg_num_dry + wgt_wet*jqg_num_wet
+                    end if
+                 end if
+   
+              end if
+   
+              denom=(log(ten))*Ze
+   
+              ! multiply ten for numerators
+              jqr  = ten*jqr_num/denom
+              jqs  = ten*jqs_num/denom
+              jqg  = ten*jqg_num/denom
+   
+           case (108)
+!          mphyopt = 108 :   Thompson MP scheme with double moment
+         
+              if ( miter == 0 ) then ! For EnKF
+                 qgh_opt = get_qgh_opt(1, 0)
+
+                 call set_dsd_para()
+                 call calcMDR()
+                 if (firstcalled) then
+                    call calcConstants()
+                    firstcalled = .false.
+                 end if
+
+                 obs_dual = init_refl()
+                 var_dsd = init_para_dsd()
+                 if (l_use_tdep_radarz) ta = T1D ! pass air temperature to radarZ
+                 call rdr_obs(real(rho, kind=r_kind), real(qscalar, kind=r_kind), obs_dual, var_dsd, 1, 1)
+                 rdBZ = real(obs_dual%T_log_ref, kind=r_kind)
+                 Ze = ten ** (rdBZ / ten)
+
+              else                   ! For Variational DA
+                 Cr=3630803456.00000_r_kind
+                 Pr=1.75_r_kind
+                 Cs_dry=958893312.000000_r_kind
+                 Cg_dry=5743808512.00000_r_kind
+                 pg_dry=1.75_r_kind
+                 Ps_dry=1.75_r_kind
+!----TM-------------
+                 Cg_wet= 5.54914E+12_r_kind
+                 Cg_dry= 1.90409E+12_r_kind
+                 Pg_wet= 2.5_r_kind
+
+!                rain
+                 Zero1 = Cr  * (rho * qrexp)**(Pr)
+                 if (  qrexp > 10E-7_r_kind .and. qscalar(P_nr) > 20._r_kind  ) then
+                    Zer = 720 *(rho*qrexp)**2_r_kind*ten**18_r_kind/(pi**2_r_kind*rhor**2_r_kind*qscalar(P_nr))
```

Comment:
Please add _r_kind to 720

Reply:
We changed the code as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-7a9c44619cb5bc5c1e48fcc0d434a8be4257007f9f4a2c8a9cd9090ff3873135R978